File: hackernews/app/schemas/summary.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel


class SummaryOut(BaseModel):
    hn_id: int
    model_version: str
    tldr: Optional[str] = None
    key_points: Optional[List[str]] = None
    consensus: Optional[str] = None
    model_name: Optional[str] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

File: my_first_app/lib/src/features/summary/domain/models/summary.dart
class Summary {
  final String hnId;
  final String? modelVersion;
  final String? modelName;
  final String? tldr;
  final List<String> keyPoints;
  final String? consensus;
  final String? createdAt;
  final String? updatedAt;

  const Summary({
    required this.hnId,
    required this.modelVersion,
    required this.modelName,
    required this.tldr,
    required this.keyPoints,
    required this.consensus,
    required this.createdAt,
    required this.updatedAt,
  });
}
